alf/algorithms/mbrl_algorithm.py

```diff
@@ -63,50 +70,66 @@ def __init__(self,
 
         Args:
             action_spec (nested BoundedTensorSpec): representing the actions.
-            dynamics_module (DDLAlgorithm): module for learning to predict
-                the next feature based on the previous feature and action.
+            dynamics_module (DynamicsLearningAlgorithm): module for learning to
+                predict the next feature based on the previous feature and action.
                 It should accept input with spec [feature_spec,
                 encoded_action_spec] and output a tensor of shape
                 feature_spec. For discrete action, encoded_action is an one-hot
                 representation of the action. For continuous action, encoded
                 action is same as the original action.
-            reward_module (REAlgorithm): module for calculating the reward,
-                i.e.,  evaluating the reward for a (s, a) pair
+            reward_module (RewardEstimationAlgorithm): module for calculating
+            the reward, i.e.,  evaluating the reward for a (s, a) pair
             planner_module (PLANAlgorithm): module for generating planned action
                 based on specified reward function and dynamics function
-            gradient_clipping (float): Norm length to clip gradients.
+            env (Environment): The environment to interact with. env is a batched
+                environment, which means that it runs multiple simulations
+                simultateously. env only needs to be provided to the root
+                Algorithm.
+            config (TrainerConfig): config for training. config only needs to be
+                provided to the algorithm which performs `train_iter()` by
+                itself.
             debug_summaries (bool): True if debug summaries should be created.
             name (str): The name of this algorithm.
 
         """
         train_state_spec = MbrlState(
-            dynamics=dynamics_module.train_state_spec, reward=(), planner=())
+            dynamics=dynamics_module.train_state_spec,
+            reward=reward_module.train_state_spec,
```

Comment:
reward state is not handled in _calc_step_reward

Reply:
Now added an interface for using reward state.